src/include/migraphx/op/prefix_scan_op.hpp

```diff
@@ -0,0 +1,101 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+#define MIGRAPHX_GUARD_OPERATORS_SCAN_OP_HPP
+
+#include <migraphx/op/name.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/shape_for_each.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+template <class Derived>
+struct prefix_scan_op : op_name<Derived>
+{
+    int64_t axis;
+    bool exclusive = false;
+    bool reverse = false;
+    
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.axis, "axis"), 
+                    f(self.exclusive, "exclusive"), 
+                    f(self.reverse, "reverse"));
+    }
+
+    value attributes() const
+    {
+        value normalize;
+        normalize["axis"] = value::array{normalize_attribute::include_min};
+        return {{"normalize_axes", normalize}};
+    }
+
+    shape normalize_compute_shape(std::vector<shape> inputs) const
+    {
+        check_shapes{inputs, *this}.has(1);
+        auto s          = inputs.at(0);
+        auto n_dims = s.lens().size();
+        if (axis >= n_dims or axis < -int64_t(n_dims)) 
+        {
+            MIGRAPHX_THROW("Axis " + std::to_string(axis) + " is out of bounds for shape with " + std::to_string(n_dims) + " dimensions");
+        }
+
+        return s;
+    }
+    
+    argument compute(const shape&, std::vector<argument> args) const
+    {
+        argument result = args[0];
+        auto s = result.get_shape();
+        auto slice = shape{s.type(), {s.lens()[axis]}, {s.strides()[axis]}};
+        auto slice_ex = shape{s.type(), {s.lens()[axis] - 1}, {s.strides()[axis]}};
+        auto lens = s.lens();
+        lens[axis] = 1;
+        auto batch = shape{s.type(), lens, s.strides()};
+        auto& self = static_cast<const Derived&>(*this);
+        result.visit([&](auto output) {
+            using type = decltype(output);
+            par_for(batch.elements(), [&](auto i) {
+                auto* start = output.data()+batch.index(i);
+                type x{slice, start};
+                if (reverse) {
+                    if (exclusive) {
+                        type y{slice_ex, start+1};
+                        type z{slice_ex, start};
+                        std::copy(y.begin(), y.end(), z.begin());
+                        *z.end() = 0;
```

Comment:
You can't deference an end iterator. This should probably be:

```cpp
std::copy(x.begin()+1, x.end(), x.begin());
x.back() = 0;
```

Reply:
I was getting errors when using pointer arithmetic on tensor_view iterators, but just tried it with pre-increment/decrement and that works.

Reply:
I see the operators for `+` are not implemented correctly.